src/gui/guiFormSpecMenu.cpp

```diff
@@ -513,14 +513,23 @@ void GUIFormSpecMenu::parseScrollBar(parserData* data, const std::string &elemen
 
 		spec.ftype = f_ScrollBar;
 		spec.send  = true;
-		gui::IGUIScrollBar* e =
-				Environment->addScrollBar(is_horizontal,rect,this,spec.fid);
+		guiScrollBar* e = new guiScrollBar(Environment, this, spec.fid, rect, is_horizontal, true);
 
 		e->setMax(data->scrollBarOptions.max);
 		e->setMin(data->scrollBarOptions.min);
 		e->setPos(stoi(parts[4]));
 		e->setSmallStep(data->scrollBarOptions.smallStep);
 		e->setLargeStep(data->scrollBarOptions.largeStep);
+		
+		if (scrollBarOptions.pageSize < 0) {
+			if (is_horizontal) {
+				e->setPageSize(dim.X);
+			} else {
+				e->setPageSize(dim.Y);
```

Comment:
Shouldn't `dim.X` and `dim.Y` be swapped?

Reply:
No, since if the scrollbar is horizontal, the page size should be the width, X.  Otherwise, it should be the height, Y.

Reply:
Oh, soory, I misunderstood pagesize.